tests/network/vmi_networking.go

```diff
@@ -167,6 +174,10 @@ var _ = SIGDescribe("[Serial][rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com]
 				tests.SkipIfOpenShift("Custom MAC addresses on pod networks are not supported")
 			}
 
+			// Re-fetch inboundVMI
+			inboundVMI, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(inboundVMI.Name, &metav1.GetOptions{})
+			Expect(err).ShouldNot(HaveOccurred())
```

Comment:
I think this belongs in the test setup, after you're sure the IP is present.

Reply:
It is actually done on the setup fixture, right?

Reply:
Right. Of course we update the VMI within the eventually block... (I think that's just wrong).

I'd rather not rely on a side effect of the eventually block (use a locally scoped variable inside the eventually block), then explicitly update the `inboundVMI` once the eventually is done.

Reply:
That is debatable.. the eventually does exactly that... refreshes the vmi until it gets what it wanted.

Lets just try to resolve this crisis as fast as possible, otherwise it will get quarantined. You can suggest a following PR to refactor this better after. Is that ok?

Reply:
In fact, a following PR should also stop using that `BeforeAll` which is known to be problematic.

Reply:
> That is debatable.. the eventually does exactly that... refreshes the vmi until it gets what it wanted.

I don't think so. A function should do one thing (that's the eventually block input parameter). That one does 2: 
  - it updates the vmi (side effect)
  - indicates that the VMI has interfaces, and that the first interface has an IP address (does the vmi has the stuff "I" need ? i.e. an IP address on the first pod interface ).

The "refreshes the vmi until it gets what it wanted" behavior you mean is brought by a function that does 2 things ran in an eventually block.

It costs seconds to rename the variable used within the eventually block, and update the VMI explicitly after it, thus preventing a side-effect (a bad code practice) from being added into the code base. If coming up with an alternative costed months (even days), I'd have kept quiet. 

> 
> Lets just try to resolve this crisis as fast as possible, otherwise it will get quarantined. You can suggest a following PR to refactor this better after. Is that ok?

I'm just trying to bring awareness that we can do the exact same without introducing side effects into the code.

Does this even matter ? Apparently something else is amiss: https://github.com/kubevirt/kubevirt/pull/5554#issuecomment-828431621.